semgrep-core/src/core/il/IL.ml

```diff
@@ -302,6 +302,13 @@ and stmt_kind =
       * (name * stmt list) list (* catches *)
       * stmt list (* finally *)
   | Throw of tok * exp (* less: enforce lval here? *)
+  | FuncStmt of {
+      fdef : G.function_definition;
```

Comment:
why not fdef: function_definition here?

Reply:
This is as above, because we need to carry the `G.function_definition` down into the IL.

Reply:
My guess is that we didn't consider doing interprocedural in semgrep for a long time, so lots of stuff like you said
was hardcoded for handling one function, and so this is why probably we had functions taking G.function_definition
instead of function_definition, but after your PR we can probably refactor the code to operate more often on a function_definition so we can drop those G.function_definition.